src/main/java/bgu/spl/net/impl/stomp/StompEncDec.java

```diff
@@ -0,0 +1,82 @@
+package bgu.spl.net.impl.stomp;
+
+import bgu.spl.net.api.MessageEncoderDecoder;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class StompEncDec implements MessageEncoderDecoder<Frame> {
+
+    private byte[] bytes = new byte[1 << 10]; //start with 1k
+    private int len = 0;
+
+    /**
+     *   Converts a String to the Frame object representing it
+     * @param msg The message as a string
+     * @return The Frame representation
+     */
+    private Frame parseMessage(String msg) {
+        int delim = 0, prevDelim = 0;
+        int msgPart = 1;
```

Comment:
why are you starting from 1 and not 0?

Reply:
Absolutely no reason at all. Comfort I suppose

Reply:
so change it to 0, and change the magical numbers